app/participants/knight.py

```diff
@@ -0,0 +1,30 @@
+from __future__ import annotations
+
+
+class Knight:
+    def __init__(self, name: str, power: int, hp: int) -> None:
+        self.name = name
+        self.power = power
+        self.hp = hp
+        self.protection = 0
+
+    def attribute_counter(
+            self, armour: list,
+            weapon: dict,
+            potion: dict | None
+    ) -> None:
+        for unit in armour:
+            self.protection += unit["protection"]
+        self.power += weapon["power"]
+        if potion:
+            for attribute, value in potion["effect"].items():
+                self.__dict__[attribute] += value
+
+    def knight_battle(self, opponent: Knight) -> None:

```

Comment:
Do not use single-standing nouns in your function names, function names should represent an action that is performed over the arguments that are passed into it. Rename this function as `fight` or `perform_battle`, etc.